rust/src/schema.rs

```diff
@@ -160,3 +160,487 @@ impl Schema {
+            { "name": "modificationTime", "type": "long", "nullable": true, "metadata": {} },
+            { "name": "dataChange", "type": "boolean", "nullable": true, "metadata": {}, },
+            { "name": "stats", "type": "string", "nullable": true, "metadata": {},
+            },/*{
+                "name": "partitionValues",
+                "type": {
+                    "type": "map",
+                    "keyType": "string",
+                    "valueType": "string",
+                    "valueContainsNull": true,
+                },
+                "nullable": true,
+                "metadata": {},
+
+            }*/]);
+
+        let mut map = HashMap::new();
+
+        map.insert(
+            "metaData".to_string(),
+            serde_json::from_value(meta_data_fields).unwrap(),
+        );
+        map.insert(
+            "protocol".to_string(),
+            serde_json::from_value(protocol_fields).unwrap(),
+        );
+        map.insert(
+            "txn".to_string(),
+            serde_json::from_value(txn_fields).unwrap(),
+        );
+        map.insert(
+            "add".to_string(),
+            serde_json::from_value(add_fields).unwrap(),
+        );
+        map.insert(
+            "remove".to_string(),
+            serde_json::from_value(remove_fields).unwrap(),
+        );
+
+        Self { common_fields: map }
+    }
+
+    /// Creates a Schema representing the delta log for a specific delta table.
+    /// Merges fields from the table schema into the delta log schema.
+    pub fn delta_log_schema_for_table(
+        &self,
+        table_schema: &Schema,
+        partition_columns: &[String],
+    ) -> Result<Schema, DeltaLogSchemaError> {
+        let (partition_fields, non_partition_fields): (Vec<SchemaField>, Vec<SchemaField>) =
+            table_schema
+                .fields
+                .iter()
+                .map(|f| f.to_owned())
+                .partition(|field| partition_columns.contains(&field.name));
+
+        let fields: Vec<SchemaField> = self
+            .common_fields
+            .iter()
+            .map(|(name, fields)| match name.as_str() {
+                "add" => {
+                    let mut fields = fields.clone();
+
+                    if !partition_fields.is_empty() {
+                        let partition_values_parsed = SchemaField {
+                            name: "partitionValues_parsed".to_string(),
+                            nullable: true,
+                            metadata: HashMap::new(),
+                            r#type: SchemaDataType::r#struct(SchemaTypeStruct {
+                                r#type: "struct".to_string(),
+                                fields: partition_fields.clone(),
+                            }),
+                        };
+                        fields.push(partition_values_parsed);
+                    }
+
+                    if !non_partition_fields.is_empty() {
```

Comment:
looks like we are missing `numRecords` here?

Reply:
fixed